acvm/src/pwg/mod.rs

```diff
@@ -154,11 +141,22 @@ impl<B: BlackBoxFunctionSolver> ACVM<B> {
         &self.witness_map
     }
 
-    /// Returns a slice containing the opcodes which remain to be solved.
-    ///
-    /// Note: this doesn't include any opcodes which are waiting on a pending foreign call.
-    pub fn unresolved_opcodes(&self) -> &[(Opcode, OpcodeLabel)] {
-        &self.opcodes_and_labels
+    /// Returns a slice containing the opcodes of the circuit being executed.
+    pub fn opcodes(&self) -> &[Opcode] {
+        &self.opcodes
+    }
+
+    /// Returns the index of the current opcode to be executed.
+    pub fn instruction_pointer(&self) -> usize {
+        self.instruction_pointer
+    }
+
+    /// Finalize the ACVM execution, returning the resulting [`WitnessMap`].
+    pub fn finalize(self) -> WitnessMap {
+        if self.status != ACVMStatus::Solved {
+            panic!("ACVM is not ready to be finalized");
```

Comment:
Can you add the status into the panic message context?

Reply:
Addressed in #455